R/ruleset_construction.R

```diff
@@ -1,24 +1,54 @@
-#' Add Rulesets Together
+#' Add Rules and Rulesets Together
 #'
-#' `+` allows you to concatenate several rulesets into a larger rulesets. This
+#' `+` allows you to add rules and rulesets into larger rulesets. This
 #' can be useful if you want to create a ruleset for a dataset out of checks
 #' for other datasets.
 #'
 #' @param a the first ruleset you wish to add
 #' @param a the second ruleset you wish to add
 #' @rdname ruleset_add
-#' @export
-"+.ruleset" <- function(a,b) {
-	out <- c(a,b)
-	class(out) <- "ruleset"
+datavarifyr_plus  <- function(a,b) {
+    # This is the method to add *both* rules and rulesets together.
+    # Has to be done this way AFAIK to comply with R's
+    # [double dispatch](https://yutani.rbind.io/post/double-dispatch-of-s3-method/)
+    # semantics.
+
+    if (class(a) == "ruleset" & class(b)=="ruleset") {
```

Comment:
what about something like this?
``R
ìf  (class(a) != "ruleset") a <- list(a)
if (class(b) != "ruleset") b <- list(b)
out <- list(a, b)
```

Reply:
This doesn't work in the case that both a and b are rules, but I like the suggestion - this could have been more concise. 

This should work: 

```r
if (class(a) == "rule" & class(b) == "rule") {
    out <- list(a,b)
} else {
    if (class(a) == "rule") a <- list(a)
    if (class(b) == "rule") b <- list(b)
    out <- c(a, b)
}
```